pkg/reconciler/apis/apiexportendpointslice/apiexportendpointslice_controller.go

```diff
@@ -227,35 +221,34 @@ func (c *controller) enqueueAPIExportEndpointSlicesForAPIExport(obj interface{})
 		} else if !exists {
 			continue
 		}
-		logger := logging.WithObject(logging.WithReconciler(klog.Background(), ControllerName), obj.(*apisv1alpha1.APIExport))
 		logging.WithQueueKey(logger, key).V(4).Info("queuing APIExportEndpointSlices because of referenced APIExport")
-		c.enqueueAPIExportEndpointSlice(slice)
+		c.enqueueAPIExportEndpointSlice(objOrTombstone[*apisv1alpha1.APIExportEndpointSlice](slice), logger, " because of referenced APIExport")
 	}
 }
 
 // enqueueAllAPIExportEndpointSlices enqueues all APIExportEndpointSlices.
-func (c *controller) enqueueAllAPIExportEndpointSlices(shard interface{}) {
+func (c *controller) enqueueAllAPIExportEndpointSlices(shard *corev1alpha1.Shard, logger logr.Logger, logSuffix string) {
```

Comment:
why react to all shards?

Reply:
Shard might change (label changed), so it got removed from partition somewhere else, so we need to recompute the partitions and selectors every time shard changed.

Reply:
I don't understand. I thought the non-url controller is for the local slices only. It does not depend (in this controller) from shards anymore.

Reply:
I think you are right. I just needed to re-read this a bit myself to understand how this is wired. And Im not sure why Shard was in the picture for this controller in the first place. 

Setup usually is like this:
```
shard1 -->
shard2 --> PartitionSet --> Partitions
shard3 --> 
```

this bubbles to downstream:
```
Partition --> APIExportEndpointSlices <-- APIExport (global as it might be from outside)
```

The non-url controller should work only on `Partitions`, `APIExport` or `APIExportEndpointSlices` changes to recompute the result. 

URL should act on `APIBindings` and `APIExportEndpointSlices`.